invenio_drafts_resources/services/records/components/base.py

```diff
@@ -200,20 +209,33 @@ def _publish_new(self, identity, draft, record):
 
     def _publish_edit(self, identity, draft, record):
         """Action when publishing an edit to an existing record."""
-        # TODO: For published records, we should sync changes from the
-        # draft bucket to the record bucket, so that an instance could
-        # potentially allow a user to update files. For now, sync() only
-        # changes the default_preview and order
-
+        # For now, sync() only # changes the default_preview and order
+        # the lock and unlock management is done here in the component
+        enforce_lock_files = self.service.config.lock_edit_published_files(record)
         record_files = self.get_record_files(record)
         draft_files = self.get_record_files(draft)
 
+        # interestingly, when we compare file manager instances, we need to have
+        # enabled files!
+        files_changed = draft_files.enabled and record_files != draft_files
+
+        record_files_bucket = record_files.bucket
+        draft_files_bucket = draft_files.bucket
+
+        # sync only metadata
         record_files.sync(draft_files)
 
+        if not enforce_lock_files and files_changed:
+            record_files.unlock()
+            draft_files_bucket.sync(bucket=record_files_bucket, delete_extras=True)
```

Comment:
@ntarocco this didn't work unfortunately, needs more testing (it does not add new files to the record after publishing)